empv.el

```diff
@@ -1585,6 +1585,22 @@ download finishes with the path downloaded."
   (interactive "sLink: ")
   (empv-youtube-download link))
 
+(defun empv-download-current-item ()
+    (interactive)
+    (let* ((link (empv--youtube-item-extract-link
+                  (or (empv-youtube-results--current-item)
+                      (car empv--last-youtube-candidates))))
+           (url (empv--clean-uri link))
+           (file-name (concat empv-audio-dir
+                              (or (string-clean-whitespace (plist-get (empv--extract-empv-metadata-from-path link) :title))
+                                  (read-string "Enter file name: "))
+                              ".mp3")))
+      (if (file-exists-p file-name)
+          (setq file-name (concat (file-name-sans-extension file-name) "-"
+                                  (nth 1 (split-string url "=")) ".mp3")))
```

Comment:
What about simply passing file-name as nil to `empv-youtube-download` so that user can interactively select a path to download?